tasks/_agent-linux-macos-shared.yml

```diff
@@ -18,36 +18,41 @@
   when: datadog_manage_config
   notify: "{{ agent_dd_notify_agent }}"
 
-- name: Register all checks directories present in datadog
+- name: Register all check configs present in datadog
   find:
     paths: "{{ agent_dd_config_dir }}/conf.d/"
     patterns:
-      - "*.d"
-    file_type: directory
-  register: agent_datadog_conf_directories
+      - "conf.yaml*"
+    recurse: true
+    depth: 2
+    file_type: file
+  register: agent_datadog_conf_paths
   when: datadog_manage_config and (datadog_disable_untracked_checks or datadog_disable_default_checks)
 
 - name: Delete checks not present in agent_datadog_tracked_checks
   file:
     path: "{{ agent_dd_config_dir }}/conf.d/{{ item }}.d/conf.yaml"
     state: absent
-  loop: "{{ agent_datadog_conf_directories.files | map(attribute='path') | list | map('basename') | list | map('regex_replace', '^(.*).d$', '\\1') | list }}"
+  loop: "{{ agent_datadog_conf_paths.files | selectattr('filename', 'eq', 'conf.yaml') | map(attribute='path') | list
```

Comment:
It looks like the `filename` attribute doesn't get defined. Following the docs on [find](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/find_module.html#return-files), the list of attributes would seem to be the one on https://docs.ansible.com/ansible/latest/collections/ansible/builtin/stat_module.html#return-stat; I don't see `filename` there. So I'm not sure this can work like that.

Reply:
My apologies, I tested on a Windows host, and just ported this over from that code. I didn't realize `find` returned different attributes than `win_find`. I'll get this work on linux and post an update.

Reply:
Should be all set now. I removed the filename dependency all together and just match the value of path instead